src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNINpHandle.cs

```diff
@@ -271,40 +268,43 @@ public override uint Send(SNIPacket packet)
                 bool releaseLock = false;
                 try
                 {
-                    // is the packet is marked out out-of-band (attention packets only) it must be
-                    // sent immediately even if a send of recieve operation is already in progress
-                    // because out of band packets are used to cancel ongoing operations
-                    // so try to take the lock if possible but continue even if it can't be taken
-                    if (packet.IsOutOfBand)
-                    {
-                        Monitor.TryEnter(this, ref releaseLock);
-                    }
-                    else
+                    lock (this)
```

Comment:
This is going to break the code below which selectively takes the lock on `this` and regress the cancelation behaviour that it fixed.

Reply:
That change was unintentional, reverted.